packages/engine/src/avatar/functions/moveAvatar.ts

```diff
@@ -133,9 +135,46 @@ export const updateReferenceSpaceFromAvatarMovement = (movement: Vector3) => {
 
 const _additionalMovement = new Vector3()
 
+/**
+ * Avatar movement via click/pointer position
+ */
+
+const minimumDistanceSquared = 0.5 * 0.5
+
+export const applyAutopilotInput = (entity: Entity) => {
+  const controller = getComponent(entity, AvatarControllerComponent)
+  if (!controller || controller.autopilotWalkpoint == undefined) return
+
+  if (controller.gamepadLocalInput.lengthSq() > 0) {
+    controller.autopilotWalkpoint = undefined
+    return
+  }
+
+  const walkpoint = new Vector3()
+  walkpoint.set(controller.autopilotWalkpoint.x, controller.autopilotWalkpoint.y, controller.autopilotWalkpoint.z)
+  const autopilotMarkerObject = markerInstance.object
+  if (autopilotMarkerObject) ScaleFluctuate(autopilotMarkerObject)
+  const avatarPos = getComponent(entity, TransformComponent).position
+  const moveDirection = walkpoint.sub(avatarPos)
+  const distanceSquared = moveDirection.lengthSq()
+  const avatarMovementSettings = getState(AvatarMovementSettingsState).value
+  const legSpeed = controller.isWalking ? avatarMovementSettings.walkSpeed : avatarMovementSettings.runSpeed
+  const delta = 0.0175
+
+  if (distanceSquared > minimumDistanceSquared)
+    updateLocalAvatarPosition(
+      moveDirection
+        .normalize()
+        .multiplyScalar(delta * legSpeed)
+        .add(new Vector3(0, controller.verticalVelocity, 0))
```

Comment:
avoid creating new vectors in hot spots. this can be `.setY(` instead of `.add(`